test/integration/reclaim_test.js

```diff
@@ -87,5 +88,39 @@ suite('Reclaiming task', function() {
       'Log file was present when there should not have been one'
     );
   });
+
+  test('test taskcluster-proxy credentials update', async function() {
+    let payload = {
+      storageType: 'reference',
+      expires: expires().toJSON(),
+      contentType: 'text/html',
+      url: 'https://mozilla.com'
+    };
+
+    let updated = false;
+    worker.on('Credentials updated', function() {
```

Comment:
[This](https://github.com/taskcluster/docker-worker/blob/master/test/integration/reclaim_test.js#L22) guarantees reclaim will happen.